### PR TITLE
:sparkles: component / camera

### PR DESCRIPTION
**Pull request by guillaumecrespel**
I've not make style.
just component and drivers logic

### CODE REVIEW

**Comment by guillaumecrespel** [comment]
@fabienjuif please, first review

**Comment by fabienjuif** [comment]
Can you fix errors please first ?
And ping me again

**Comment by guillaumecrespel** [comment]
it's good ;) @fabienjuif

File: ui/ui-expenses/src/screens/list/list.reactions.js [hunk 1]
```diff
@@ -56,6 +56,7 @@ export const submit = ({ payload }, store, { window, http }) => {
           const reader2 = new window.FileReader()
           reader2.onerror = (readerE) => { store.dispatch({ type: '@@file/ON_ERROR', payload: readerE }) }
           reader2.onload = async (readerE) => {
+            store.devices.cameras.capture.set(readerE.target.result)
```
Comment (fabienjuif) [inline ui/ui-expenses/src/screens/list/list.reactions.js]: what's the purpose ?

File: ui/ui-expenses/src/screens/prices/capture/capture.container.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import Component from './capture'
+
+
+const mapStore = (store, ownProps, { camera, window }) => ({
+  device: camera.getCameras()[0],
+  capture: camera.getCapture(),
+  useCamera: camera.useCamera,
+  takeCapture: camera.takeCapture,
+  cancel: camera.removeCapture,
+  height: window.innerHeight / 3,
```
Comment (fabienjuif) [inline ui/ui-expenses/src/screens/prices/capture/capture.container.js]: If you don't make styles, remove this.
This is bad practice

File: ui/ui-expenses/src/screens/prices/capture/capture.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Webcam from 'react-webcam'
+import { component } from 'ui-hocs'
+
+const Capture = ({
+  device,
+  height,
```
Comment (fabienjuif) [inline ui/ui-expenses/src/screens/prices/capture/capture.jsx]: to remove
 - height,
 - width

File: ui/ui-expenses/src/screens/prices/capture/capture.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    )
+  }
+  return (
+    <div>
+      <Webcam
+        audio={false}
+        height={height}
+        width={width}
+        ref={useCamera}
+        screenshotFormat="image/webp"
```
Comment (fabienjuif) [inline ui/ui-expenses/src/screens/prices/capture/capture.jsx]: Can you save to jpeg, so we can process the image like we do with screenshot etc

File: ui/ui-expenses/src/screens/prices/prices.jsx [hunk 5]
```diff
@@ -35,38 +37,44 @@ const Prices = ({
...
     <div className={styles.block}>
-      <h2>Prix</h2>
-      <div className={styles.buttons}>
-        {prices.map(price => (
-          <Price key={price} price={price} selected={price === prices[0]} />
-        ))}
-      </div>
+      <Capture />
     </div>
+    {prices.length > 0 &&
```
Comment (fabienjuif) [inline ui/ui-expenses/src/screens/prices/prices.jsx]: Please no.
You can do a route or something.

Comment (fabienjuif) [reply ui/ui-expenses/src/screens/prices/prices.jsx]: or use the loader
or other solution than that

File: ui/ui-expenses/src/store/drivers/camera.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      list: simpleObject({ defaultData: [] }),
+      capture: simpleObject({ defaultData: '' }),
+      googleId: simpleObject({ defaultData: '' }),
+    },
+  }),
+  getDriver: (store) => {
+    store.listeners.add([
+      when('@@krml/INIT')(setDevices),
+      when('@@krf/SET>DEVICES_CAMERAS>CAPTURE')(sendOnDrive),
+      when('@@http/IMAGES>POST>ENDED')(setGoogleId),
```
Comment (fabienjuif) [inline ui/ui-expenses/src/store/drivers/camera.js]: You should not set this kind of event in the driver.
This is screens logics.

Drivers are not something you put all in.
This is like `drivers` from cyclejs.

When you write drivers think about this : is it reusable in an other project ?

I think the event I comment and nexts should not be there

File: ui/ui-expenses/src/screens/list/list.reactions.js (hunk 1 above)
Comment (guillaumecrespel) [reply ui/ui-expenses/src/screens/list/list.reactions.js]: to store capture for viewing on next screen.

File: ui/ui-expenses/src/screens/prices/capture/capture.container.js (hunk 2 above)
Comment (guillaumecrespel) [reply ui/ui-expenses/src/screens/prices/capture/capture.container.js]: this props it's not only to style.
The component `Capture` use a canva, and this size is to define the size of capture. Not just the size of component.

File: ui/ui-expenses/src/screens/prices/capture/capture.jsx (hunk 3 above)
Comment (guillaumecrespel) [reply ui/ui-expenses/src/screens/prices/capture/capture.jsx]: see previous comment

File: ui/ui-expenses/src/screens/prices/capture/capture.jsx (hunk 4 above)
Comment (guillaumecrespel) [reply ui/ui-expenses/src/screens/prices/capture/capture.jsx]: ok, just to see the size of picture with this format

File: ui/ui-expenses/src/screens/prices/prices.jsx (hunk 5 above)
Comment (guillaumecrespel) [reply ui/ui-expenses/src/screens/prices/prices.jsx]: in my mind is temporary, the `Price` screen will be refactor after.
But ok, I delete this condition.

**Pull request closed without merging**